pennylane/measurements/mid_measure.py

```diff
@@ -85,12 +103,13 @@ def func():
     Args:
         wires (Wires): The wire of the qubit the measurement process applies to.
         reset (Optional[bool]): Whether to reset the wire to the :math:`|0 \rangle`
-            state after measurement.
+            state after measurement. It should be `False` when used with :func:`~.qjit`.
```

Comment:
We should be able to implement reset easily for users by adding `cond(m)(PauliX)` after our measure call.

Reply:
@dime10 yeah, what do you think of implementing this feature in Catalyst for the next release? It also makes the func signatures of `qml.measure` and `catalyst.measure` consistent.

Reply:
Good idea, we can add it to catalyst as well 👍 But I think in the meantime it's two lines to add to the dispatching function, and would get rid of the additional comments in the docstring.